tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-tree.ts

```diff
@@ -0,0 +1,108 @@
+import { capitalize } from "./capitalize";
+import { sortedArrayInsert } from "./sorted-array-insert";
+
+export interface NavigationDocument {
+  id: string;
+  title: string;
+  path: string;
+}
+
+/**
+ * A class to form a navigation Prefix Trie from a list of navigation documents
+ * where each documents id represents a path that is 
+ * separated by the `/` character.
+ */
+export class NavigationTree {
+  id: string = '';
+  path?: string;
+  title: string = '';
+  children: NavigationTree[] = [];
+
+  // Add the document to the `documents` of the relevant child
+  //Check to see if there's a parent document that needs to moved down,
+  //If there is, shift it down.
+  constructor(key: string = '', title: string = '', path: string = '', children: NavigationTree[] = []) {
+    this.id = key;
+    this.title = title;
+    if(path){
+      this.path = path;
+    }
+    this.children = children;
+  }
+
+  static generateTreeFromDocuments(items: NavigationDocument[] = []) {
```

Comment:
I feel like this is a weird function for the `NavigationTree` to have. I'd rather it create a new tree upon construction, then maybe have an `insertMany` function.

Reply:
I agree. I felt this function was odd as a static member. I was considering it as part of the constructor, but it felt too complex. Do you think it makes sense as a separate pure function?

Reply:
I've moved this into a pure function.